src/qubx/data/helpers.py

```diff
@@ -189,23 +190,27 @@ def _handle_symbols_data_from_to(
         if _new_symbols:
             _s_req = min(_start, self._start if self._start else _start)
             _e_req = max(_stop, self._stop if self._stop else _stop)
-            logger.debug(f"Loading all data {_s_req} - {_e_req} for { ','.join(_new_symbols)} ")
-            _new_data = self._load_candle_data(_new_symbols, _s_req, _e_req + _dtf, self._data_timeframe)
+            logger.debug(f"(InMemoryCachedReader) Loading all data {_s_req} - {_e_req} for { ','.join(_new_symbols)} ")
+            # _new_data = self._load_candle_data(_new_symbols, _s_req, _e_req + _dtf, self._data_timeframe)
+            _new_data = self._load_candle_data(_new_symbols, _s_req, _e_req, self._data_timeframe)
             self._data |= _new_data
 
-        # - part intervals
+        # - pre intervals
         if self._start and _start < self._start:
             _smbs = list(self._data.keys())
-            logger.debug(f"Updating {len(_smbs)} symbols before interval {_start} : {self._start}")
-            _before = self._load_candle_data(_smbs, _start, self._start + _dtf, self._data_timeframe)
+            logger.debug(f"(InMemoryCachedReader) Updating {len(_smbs)} symbols pre interval {_start} : {self._start}")
+            # _before = self._load_candle_data(_smbs, _start, self._start + _dtf, self._data_timeframe)
+            _before = self._load_candle_data(_smbs, _start, self._start, self._data_timeframe)
             for k, c in _before.items():
-                self._data[k] = srows(c, self._data[k], keep="first")
+                # self._data[k] = srows(c, self._data[k], keep="first")
+                self._data[k] = srows(c, self._data[k], keep="last")
 
-        # - part intervals
+        # - post intervals
         if self._stop and _stop > self._stop:
             _smbs = list(self._data.keys())
-            logger.debug(f"Updating {len(_smbs)} symbols after interval {self._stop} : {_stop}")
-            _after = self._load_candle_data(_smbs, self._stop - _dtf, _stop, self._data_timeframe)
+            logger.debug(f"(InMemoryCachedReader) Updating {len(_smbs)} symbols post interval {self._stop} : {_stop}")
```

Comment:
I suggest adding class name into the logger configuration. But let's do this later in a separate refactoring.